src/layout_configuration.cpp

```diff
@@ -0,0 +1,98 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "layout_configuration.hpp"
+
+#include <algorithm>
+#include <sstream>
+#include <vector>
+
+#include "stringutils.hpp"
+
+namespace ovms {
+
+LayoutConfiguration::LayoutConfiguration(const char* layout) :
+    LayoutConfiguration(std::string(layout)) {
+}
+
+LayoutConfiguration::LayoutConfiguration(const std::string& layout) :
+    LayoutConfiguration(layout, layout) {
+}
+
+LayoutConfiguration::LayoutConfiguration(const std::string& tensorLayout, const std::string& modelLayout) :
+    tensor(tensorLayout),
+    model(modelLayout) {
+}
+
+bool LayoutConfiguration::isSet() const {
+    return !tensor.empty() || !model.empty();
+}
+
+Status LayoutConfiguration::fromString(const std::string& configurationStr, LayoutConfiguration& configOut) {
+    std::string configurationCopy = configurationStr;
+    erase_spaces(configurationCopy);
+
+    std::transform(configurationCopy.begin(), configurationCopy.end(), configurationCopy.begin(), ::toupper);
+
+    if (configurationCopy.find_first_not_of("NCHWD?.:") != std::string::npos)
+        return StatusCode::LAYOUT_WRONG_FORMAT;
```

Comment:
Maybe we could define
const char LAYOUT_CONFIG_DELIMITER = ":";
const std::string ALLOWED_LAYOUT_CHARS = "NCHWD?.";
const  std::string ALLOWED_LAYOUT_CONFIG_CHARS = ALLOWED_LAYOUT_CHARS + LAYOUT_CONFIG_DELIMITER;
etc?